### ISSUE TITLE
one pic error!... (Without change of Batch size)

### ISSUE THREAD

**Issue by ALK416** [carpedm20/DCGAN-tensorflow#353]
I run this DCGAN in colab and I've only changed a few parameters(shown below).
The model was successfully lunched and the training process is good.
But there are " one pic error " every time the sample pic should be saved, so I got no images in the output folder.

Parameter changed: 
"input_height", 96
"input_width", 96
"output_height", 48
"output_width", 48
"train", True
"crop", True
"visualize", True
"sample_freq", 10

Anyone who know how to solve this problem? Thanks for help.

**Comment by ALK416** [comment]
Also there is a warning:

/usr/local/lib/python2.7/dist-packages/ipykernel_launcher.py:102: DeprecationWarning: `imresize` is deprecated!
`imresize` is deprecated in SciPy 1.0.0, and will be removed in 1.3.0.
Use Pillow instead: ``numpy.array(Image.fromarray(arr).resize())``.

**Comment by zhao181** [comment]
> I run this DCGAN in colab and I've only changed a few parameters(shown below).
> The model was successfully lunched and the training process is good.
> But there are " one pic error " every time the sample pic should be saved, so I got no images in the output folder.
> 
> Parameter changed:
> "input_height", 96
> "input_width", 96
> "output_height", 48
> "output_width", 48
> "train", True
> "crop", True
> "visualize", True
> "sample_freq", 10
> 
> Anyone who know how to solve this problem? Thanks for help.


How did you solve this? I met the same 'one pic error!' problem

**Comment by Heisenberg1997** [comment]
> > I run this DCGAN in colab and I've only changed a few parameters(shown below).
> > The model was successfully lunched and the training process is good.
> > But there are " one pic error " every time the sample pic should be saved, so I got no images in the output folder.
> > Parameter changed:
> > "input_height", 96
> > "input_width", 96
> > "output_height", 48
> > "output_width", 48
> > "train", True
> > "crop", True
> > "visualize", True
> > "sample_freq", 10
> > Anyone who know how to solve this problem? Thanks for help.
> 
> How did you solve this? I met the same 'one pic error!' problem
How did you solve this? I met the same 'one pic error!' problem

**Comment by ALK416** [comment]
I solved this problem a long time ago so I barely remember the detail of this problem.
I finally run the program successfully and record the final code as following :


python main.py --input_height 96 --input_width 96 --output_height 48 --output_width 48 --dataset anime --crop --train --ckpt_freq 5 --sample_freq 4 visualize=True&nbsp; --epoch 200 --input_fname_pattern "*.jpg"


I hope that's useful to you.





------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Heisenberg Joker"<notifications@github.com&gt;;
发送时间:&nbsp;2020年2月15日(星期六) 晚上9:54
收件人:&nbsp;"carpedm20/DCGAN-tensorflow"<DCGAN-tensorflow@noreply.github.com&gt;;
抄送:&nbsp;"909152778"<szh1158985513@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [carpedm20/DCGAN-tensorflow] one pic error!... (Without change of Batch size) (#353)



  
I run this DCGAN in colab and I've only changed a few parameters(shown below).
 The model was successfully lunched and the training process is good.
 But there are " one pic error " every time the sample pic should be saved, so I got no images in the output folder.
 Parameter changed:
 "input_height", 96
 "input_width", 96
 "output_height", 48
 "output_width", 48
 "train", True
 "crop", True
 "visualize", True
 "sample_freq", 10
 Anyone who know how to solve this problem? Thanks for help.
  
How did you solve this? I met the same 'one pic error!' problem
 How did you solve this? I met the same 'one pic error!' problem

**Comment by Heisenberg1997** [comment]
> I solved this problem a long time ago so I barely remember the detail of this problem. I finally run the program successfully and record the final code as following : python main.py --input_height 96 --input_width 96 --output_height 48 --output_width 48 --dataset anime --crop --train --ckpt_freq 5 --sample_freq 4 visualize=True&nbsp; --epoch 200 --input_fname_pattern "*.jpg" I hope that's useful to you.
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人:&nbsp;"Heisenberg Joker"<notifications@github.com&gt;; 发送时间:&nbsp;2020年2月15日(星期六) 晚上9:54 收件人:&nbsp;"carpedm20/DCGAN-tensorflow"<DCGAN-tensorflow@noreply.github.com&gt;; 抄送:&nbsp;"909152778"<szh1158985513@qq.com&gt;;"Author"<author@noreply.github.com&gt;; 主题:&nbsp;Re: [carpedm20/DCGAN-tensorflow] one pic error!... (Without change of Batch size) (#353) I run this DCGAN in colab and I've only changed a few parameters(shown below). The model was successfully lunched and the training process is good. But there are " one pic error " every time the sample pic should be saved, so I got no images in the output folder. Parameter changed: "input_height", 96 "input_width", 96 "output_height", 48 "output_width", 48 "train", True "crop", True "visualize", True "sample_freq", 10 Anyone who know how to solve this problem? Thanks for help. How did you solve this? I met the same 'one pic error!' problem How did you solve this? I met the same 'one pic error!' problem — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

Can you send me a full code via email?  It still can not work. Maybe I change my code. My email is 1415637345@qq.com  or heisenbergjoker97@gmail.com thank you!!!

**Comment by vortextemporum** [comment]
Some problem! Whatever I tried to do, adjust the settings, I always get the one pic error :(

**Comment by u19900101** [comment]
> Same problem! Whatever I tried to do, adjust the settings, I always get the one pic error :(

have you solved it?

**Comment by TsumugiWenders** [comment]
You can try to modify the imsave function in the utils.py file
```
import imageio

def imsave(images, size, path):
  image = np.squeeze(merge(images, size))
  return  imageio.imwrite(path,image)
```
it also some

**Comment by keepwith** [comment]
> You can try to modify the imsave function in the utils.py file
> 
> ```
> import imageio
> 
> def imsave(images, size, path):
>   image = np.squeeze(merge(images, size))
>   return  imageio.imwrite(path,image)
> ```
> 
> it also some

yeah, thanks for your codes, I find since scipy.misc removed imsave imread ,etc since scipy version 1.13 , so when use scipy.misc.imsave, this problem will arise,  we can use other libs with the function of save images instead, like matplotlib.pyplot, imageio etc.
